### PR TITLE
Add support for junit test out

### PR DESCRIPTION
**Pull request by jstourac**
I have added support to generate results of test also in JUnit format to easily connect this testsuite e.g. with Jenkins.

By default file in current directory named 'report.xml' is created. User can change this default with '-j' option. Please provide your opinion and possible suggestions.

### CODE REVIEW

**Comment by summerwind** [comment]
Thank you for the contribution! I think that JUnit output is very useful.
I have one feedback. I'd like to output report.xml file only if `-j` flag is enabled. Could you change that way?

**Comment by jstourac** [comment]
Hi there, I have changed code behaviour to reflect your suggestion in latest commit. Let me know whether you are ok with that change and I'll squash latest commit into previous one.

If you have any other concerns, just let me know.

File: h2spec.go [hunk 1]
```diff
@@ -248,7 +262,7 @@ func (tc *TestCase) PrintEphemeralDesc() {
 func (tc *TestCase) PrintPass() {
 	mark := "✓"
 	logger.Clear()
-	logger.Write("\x1b[32m%s\x1b[0m \x1b[90m%s\x1b[0m\n", mark, tc.Desc)
+	logger.Write("\x1b[32m%s\x1b[0m \x1b[90m%s; test time: %s\x1b[0m\n", mark, tc.Desc, tc.testTime)
```
Comment (summerwind) [inline h2spec.go]: Personally, I think that it's unnecessary because this is complicated to show the description.

File: h2spec.go [hunk 2]
```diff
@@ -257,7 +271,7 @@ func (tc *TestCase) PrintFail(expected []Result, actual Result) {
 	logger.Clear()
 
 	logger.SetColor("red")
-	logger.Write("%s %s\n", mark, tc.Desc)
+	logger.Write("%s %s; test time: %s\n", mark, tc.Desc, tc.testTime)
```
Comment (summerwind) [inline h2spec.go]: ditto.

**Comment by summerwind** [comment]
Thank you for the fix!
I put few comments. everything else is OK!

**Comment by summerwind** [comment]
Thank you for the fix. Looks good to me and let's merge.
I'm going to publish the new version soon.

**Comment by jstourac** [comment]
Thanks! Now I realized that I did not provide appropriate change to README.MD file. Shall I provide or will you cover it already? No problem for me, just asking.

**Comment by summerwind** [comment]
Oh! I'll update README.md before publishing the release. Thank you for following me!

**Comment by summerwind** [comment]
Released in v1.4.0. Thank you!
https://github.com/summerwind/h2spec/releases/tag/v1.4.0

**Pull request merged**